pkg/stack/unwind/compact_unwind_table.go

```diff
@@ -290,23 +295,71 @@ func CompactUnwindTableRepresentation(unwindTable UnwindTable, arch elf.Machine)
 
 // GenerateCompactUnwindTable produces the compact unwind table for a given
 // executable.
-func GenerateCompactUnwindTable(file *objectfile.ObjectFile) (CompactUnwindTable, elf.Machine, frame.FrameDescriptionEntries, error) {
+func GenerateCompactUnwindTable(file *objectfile.ObjectFile, logger log.Logger, debugFrameErrors prometheus.Counter) (CompactUnwindTable, elf.Machine, frame.FrameDescriptionEntries, error) {
 	var ut CompactUnwindTable
+	isUnexpected := func(err error) bool {
+		return err != nil && !errors.Is(err, ErrEhFrameSectionNotFound) && !errors.Is(err, ErrNoFDEsFound)
+	}
 
 	// Fetch FDEs.
-	fdes, arch, err := ReadFDEs(file)
-	if err != nil {
-		return ut, arch, fdes, err
+	ehFrameFdes, arch, err := ReadFDEs(file, true)
```

Comment:
I'm not a fan of boolean parameters.  Also I think the extra .debug_frame reading should be buried below the ReadFDEs function.   Seems unnecessary to complicate higher level layers with that complexity.